src/dune_rules/dune_file.ml

```diff
@@ -2332,15 +2332,27 @@ type t =
   ; stanzas : Stanzas.t
   }
 
+let is_promoted_rule version rule =
+  let is_promoted_mode = function
+    | Rule.Mode.Promote { only = None; lifetime; _ } ->
+      if version >= (3, 4) then
```

Comment:
`version >= (3, 4) || (match lifetime with Unlimited -> true | Until_clean -> false)`

Reply:
That would be `not (version >= ...) || match ...` so at that point I think that the explicit `if` is clearer.